src/ArborX_LinearBVH.hpp

```diff
@@ -132,23 +136,58 @@ class BoundingVolumeHierarchy
 
   size_t _size;
   bounding_volume_type _bounds;
-  Kokkos::View<Node *, DeviceType> _internal_and_leaf_nodes;
+  Kokkos::View<Node *, MemorySpace> _internal_and_leaf_nodes;
 };
 
-template <typename DeviceType>
-using BVH = BoundingVolumeHierarchy<DeviceType>;
+namespace Details
+{
+template <typename T>
+struct is_device_type_impl : std::false_type
+{
+};
+template <typename ExecutionSpace, typename MemorySpace>
+struct is_device_type_impl<Kokkos::Device<ExecutionSpace, MemorySpace>>
+    : std::true_type
+{
+};
+template <typename T>
+using is_device_type = typename is_device_type_impl<std::remove_cv_t<T>>::type;
+} // namespace Details
 
 template <typename DeviceType>
-template <typename Primitives>
-BoundingVolumeHierarchy<DeviceType>::BoundingVolumeHierarchy(
-    Primitives const &primitives)
+class BoundingVolumeHierarchy<
+    DeviceType, std::enable_if_t<Details::is_device_type<DeviceType>::value>>
+    : public BoundingVolumeHierarchy<typename DeviceType::memory_space>
+{
+public:
+  using device_type = DeviceType;
+  BoundingVolumeHierarchy() = default;
```

Comment:
See above. Breaking lines after multiline declarations improves readability in my opinion.

Reply:
Don't care, not breaking any line in that definition, this is code that will be deprecated and removed later.

Reply:
Done

Reply:
Then just deprecate the specialization already.

Reply:
Premature, hasn't been updated throughout ArborX yet